clusterctl/validation/validate_cluster_api_objects_test.go

```diff
@@ -0,0 +1,242 @@
+/*
```

Comment:
The tests in this file do a good job of verifying the error cases, however, validate_cluster_api_objects.go has a side effect which is a lot of string output. We should have tests that verify this output and catch when changes are made to the output. The 'fmt' library has methods which take an io.Writer. We should change validate_cluster_api_objects to take an io.Writer.

The CLI command can pass in os.Stdout and potentially os.Stderr as well if we want to support writing to a regular output and an error output. The tests would pass in a bytes.Buffer which can be used in place of the io.Writer interface: https://golang.org/pkg/bytes/#Buffer.Write. The buffer can be converted to string and then the output can be compared against what is expected. The expected output should probably be checked into golden files in the testdata folder so that we can reason about the user experience of this command / file.

Please add tests that verify the fmt output of at least two scenarios:
1. Success (no errors / FAILs)
2. Errors (at least 2 errors / FAILS in the output).